### DISCUSSION TITLE
Atom-wise contributions to CrippenLogP, CrippenMR, TPSA, LabuteASA in python

### DISCUSSION THREAD

**Discussion by mane292** [rdkit/rdkit#5352] [General] [answered]
Hi all,

I would like to calculate the contributions of each atom in a molecule to the CrippenlogP, CrippenMR, TPSA and Labute approximate surface area values. While the molecule totals are available via python I can't find an easy way to determine the atom-specific values.  It looks like this functionality is only available in the C++ API,  is there any easy way to  access the atom-wise values from the python API or do I need to use C++?

**Comment by greglandrum** [comment, marked as answer]
@mane292 the function `rdMolDescriptors._CalcCrippenContribs(mol)` returns a list of 2-tuples which provides the contributions of each atom to logp and MR.
`rdMolDescriptors._CalcLabuteASAContribs(mol)` returns a 2-tuple where the first element is the atomic contributions and the second is the contributions of all implicit Hs
Similarity `rdMolDescriptors._CalcTPSAContribs(mol)` returns a tuple with the contribution of each atom to the TPSA.

**Reply by mane292** [reply]
Wonderful, thank you!
